### PR TITLE
Expose and test string implicit cast to ReadOnlySpan<char>

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/25413
Depends on https://github.com/dotnet/coreclr/pull/15167
Depends on https://github.com/dotnet/corert/pull/4990
cc: @jkotas, @ahsonkhan

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Runtime/ref/System.Runtime.cs [hunk 1]
```diff
@@ -2322,6 +2322,7 @@ public sealed partial class String : System.Collections.Generic.IEnumerable<char
         public string TrimStart() { throw null; }
         public string TrimStart(char trimChar) { throw null; }
         public string TrimStart(params char[] trimChars) { throw null; }
+        public static implicit operator System.ReadOnlySpan<char>(string value) { throw null; }
```
Comment (ahsonkhan) [inline src/System.Runtime/ref/System.Runtime.cs]: Are you planning to add the API to System.Memory?

**Comment by stephentoub** [comment]
> Are you planning to add the API to System.Memory?

@ahsonkhan, it needs to be on string, so can't be added to System.Memory.

**Pull request merged**